modules/index/src/main/java/org/apache/ignite/internal/index/IndexBuildController.java

```diff
@@ -141,12 +142,17 @@ private CompletableFuture<?> onIndexBuilding(StartBuildingIndexEventParameters p
 
             var startBuildIndexFutures = new ArrayList<CompletableFuture<?>>();
 
-            for (TablePartitionId primaryReplicaId : primaryReplicaIds) {
-                if (primaryReplicaId.tableId() == indexDescriptor.tableId()) {
-                    CompletableFuture<?> startBuildIndexFuture = getMvTableStorageFuture(parameters.causalityToken(), primaryReplicaId)
-                            .thenCompose(mvTableStorage -> awaitPrimaryReplica(primaryReplicaId, clockService.now())
+            for (ZonePartitionId zonePartitionId : primaryReplicaIds) {
+
+                int tableId = zonePartitionId.tableId();
+
+                TablePartitionId tablePartId = new TablePartitionId(tableId, zonePartitionId.partitionId());
+
+                if (tableId == indexDescriptor.tableId()) {
```

Comment:
I guess, when we will have one partition for all tables within a zone, we will have one index as well. So, this will be changed to 
```
if (zoneId == indexDescriptor.zoneId()) {
```